rest/route/createhost_routes.go

```diff
@@ -4,16 +4,85 @@ import (
 	"context"
 	"net/http"
 
+	"github.com/evergreen-ci/evergreen/apimodels"
+	dbModel "github.com/evergreen-ci/evergreen/model"
 	"github.com/evergreen-ci/evergreen/rest"
 	"github.com/evergreen-ci/evergreen/rest/data"
 	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/evergreen/util"
 	"github.com/evergreen-ci/gimlet"
+	"github.com/gorilla/mux"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
 )
 
+type createHostHandler struct {
+	taskID     string
+	createHost apimodels.CreateHost
+}
+
 type listCreateHostHandler struct {
 	taskID string
 }
 
+func getCreateHostRouteManager(route string, version int) *RouteManager {
+	return &RouteManager{
+		Route: route,
+		Methods: []MethodHandler{
+			{
+				Authenticator:  &NoAuthAuthenticator{},
+				RequestHandler: &createHostHandler{},
+				MethodType:     http.MethodPost,
+			},
+		},
+	}
+}
+
+func (h *createHostHandler) Handler() RequestHandler {
+	return &createHostHandler{}
+}
+
+func (h *createHostHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	taskID := mux.Vars(r)["task_id"]
```

Comment:
use gimlet.GetVars() I'm trying to pull all references to mux so I can pull the dependency entirely.